gtest/src/manager.rs

```diff
@@ -4,17 +4,28 @@ use gear_core::{
     message::{Message, MessageId},
     program::{Program, ProgramId},
 };
+use std::cell::RefCell;
 use std::collections::{BTreeMap, VecDeque};
 
+use crate::check::ProgramInitializer;
+
 #[derive(Clone, Default)]
 pub struct InMemoryExtManager {
     message_queue: VecDeque<Message>,
     log: Vec<Message>,
-    programs: BTreeMap<ProgramId, Program>,
+    programs: RefCell<BTreeMap<ProgramId, Program>>,
```

Comment:
Since the program is stored before any processing now I had to repeat the logic in the gtest/node test suite, so `ProgramStorage` trait is introduced. It has only one method with signature `&self`.